packages/react-avatar/src/components/Avatar/useAvatar.tsx

```diff
@@ -78,17 +71,17 @@ export const useAvatar = (props: AvatarProps, ref: React.Ref<HTMLElement>): Avat
 
 const getDefaultIcon = (size: NonNullable<AvatarProps['size']>) => {
   if (size <= 24) {
-    return <Person16Regular />;
+    return <PersonRegular fontSize={16} />;
   } else if (size <= 40) {
-    return <Person20Regular />;
+    return <PersonRegular fontSize={20} />;
   } else if (size <= 48) {
-    return <Person24Regular />;
+    return <PersonRegular fontSize={24} />;
   } else if (size <= 56) {
-    return <Person28Regular />;
+    return <PersonRegular fontSize={28} />;
   } else if (size <= 72) {
-    return <Person32Regular />;
+    return <PersonRegular fontSize={32} />;
   } else {
-    return <Person48Regular />;
+    return <PersonRegular fontSize={48} />;
   }
```

Comment:
I think I'd prefer to do this in a different way for Avatar. Rather than describe the approach and ask you to implement it, I just made a PR here: #21160. Could you review/approve that PR when you have a chance, and revert the Avatar changes in this PR? Thanks!

Reply:
Wanted to comment the same, but @behowell was first 😉

---

With an explicit `fontSize` it becomes really strange, for example:

```tsx
function App() {
  return (
    <>
      {/* 🟡 This is kinda okay, but what is proper value for `fontSize`? How can guess it? */}
      <Component icon={<FooRegular fontSize={20} />} />
      {/* 🔴 "Component" uses `size` */}
      {/* It will be a trial for customers to know mapping between `size` & `fontSize` */}
      <Component
        size="large"
        icon={
          <FooRegular fontSize={20 /* "20" is small, should I use "40"? */} />
        }
      />
    </>
  );
}
```

We should not ask users to define `fontSize` explicitly when icons are used with our components. IMO the ideal usage is:

```tsx
function App() {
  return <Component icon={<FooRegular />} />;
}
```